### PR TITLE
[ADF-4264] Should update the breadcrumb current folder to chosen location - take2

### PR DESCRIPTION
**Pull request by gmandakini**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://issues.alfresco.com/jira/browse/ADF-4264

### CODE REVIEW

File: e2e/content-services/custom-site-list.e2e.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+        await this.alfrescoJsApi.login(TestConfig.adf.adminEmail, TestConfig.adf.adminPassword);
+
+        await this.alfrescoJsApi.core.peopleApi.addPerson(acsUser);
+
+        await this.alfrescoJsApi.login(acsUser.id, acsUser.password);
+
+        loginPage.loginToContentServicesUsingUserModel(acsUser);
+        navigationBarPage.clickNodeSelectorButton();
+        browser.ignoreSynchronization = true;
```
Comment (cristinaj) [inline e2e/content-services/custom-site-list.e2e.ts]: Is this needed?

File: e2e/pages/adf/demo-shell/node-selector/objectPicker.page.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    breadCrumbDropdown = new BreadCrumbDropdownPage(this.customSiteListPanel);
+    sitesDropdown = new SitesDropdownPage(this.customSiteListPanel);
+    contentList = new DocumentListPage(this.customSiteListPanel);
+
+    checkObjectPickerTitleIsDisplayed() {
+        BrowserVisibility.waitUntilElementIsVisible(this.title);
+    }
+
+    clickCustomSiteListPanel() {
+        BrowserVisibility.waitUntilElementIsVisible(this.customSiteListPanel);
```
Comment (eromano) [inline e2e/pages/adf/demo-shell/node-selector/objectPicker.page.ts]: please use BrowserActions.click

File: e2e/pages/adf/demo-shell/node-selector/objectPicker.page.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        return this.customSiteListPanel.click();
+    }
+
+    checkCustomSiteListExpanded() {
+        BrowserVisibility.waitUntilElementIsVisible(this.customSiteListExpanded);
+    }
+
+    enterTextInSiteGuid(text) {
+        BrowserVisibility.waitUntilElementIsVisible(this.siteGuid);
```
Comment (eromano) [inline e2e/pages/adf/demo-shell/node-selector/objectPicker.page.ts]: please use BrowserActions.clearSendKeys

File: e2e/pages/adf/demo-shell/node-selector/objectPicker.page.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        return this.siteGuid.sendKeys(text);
+    }
+
+    enterTextInSiteTitle(text) {
+        BrowserVisibility.waitUntilElementIsVisible(this.siteTitle);
```
Comment (eromano) [inline e2e/pages/adf/demo-shell/node-selector/objectPicker.page.ts]: please use BrowserActions.clearSendKeys

File: e2e/pages/adf/demo-shell/node-selector/objectPicker.page.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        return this.siteTitle.sendKeys(text);
+    }
+
+    clickAddButton() {
+        BrowserVisibility.waitUntilElementIsVisible(this.addButton);
```
Comment (eromano) [inline e2e/pages/adf/demo-shell/node-selector/objectPicker.page.ts]: please use BrowserActions.clearSendKeys
and BrowserActions.click

File: e2e/content-services/custom-site-list.e2e.ts (hunk 1 above)
Comment (gmandakini) [reply e2e/content-services/custom-site-list.e2e.ts]: yes, without this the test is failing to run.

File: e2e/pages/adf/demo-shell/node-selector/objectPicker.page.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { BrowserVisibility, BreadCrumbDropdownPage, SitesDropdownPage, DocumentListPage, BrowserActions } from '@alfresco/adf-testing';
+import { element, by } from 'protractor';
+
+export class ObjectPickerPage {
+
+    title = element(by.cssContainingText('ng-component main h1', 'Object picker'));
```
Comment (marouanbentaleb) [inline e2e/pages/adf/demo-shell/node-selector/objectPicker.page.ts]: Can't we use another locator here ?

File: e2e/content-services/custom-site-list.e2e.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        objectPickerPage.checkCustomSiteListExpanded();
+        objectPickerPage.enterTextInSiteGuid('-mysites-');
+        objectPickerPage.enterTextInSiteTitle('File Libraries');
+        objectPickerPage.clickAddButton();
+        objectPickerPage.contentListPage().dataTablePage().doubleClickRowByContent('Builds');
+        objectPickerPage.contentListPage().dataTablePage().tableIsLoaded();
+        done();
+    });
+
+    // skipping the test as known issue exists ADF-4264
```
Comment (marouanbentaleb) [inline e2e/content-services/custom-site-list.e2e.ts]: We need to make sure this test is enabled when ADF-4264 is done.

Comment (gmandakini) [reply e2e/content-services/custom-site-list.e2e.ts]: yes, will be enabled when the issue is fixed.

Comment (eromano) [inline e2e/content-services/custom-site-list.e2e.ts]: Please don't add those comment in the code. if is not possible to add this test because of a bug move it in a separate PR and put it in hold with the tag waiting for dev fix

File: e2e/pages/adf/demo-shell/node-selector/objectPicker.page.ts (hunk 6 above)
Comment (gmandakini) [reply e2e/pages/adf/demo-shell/node-selector/objectPicker.page.ts]: any suggestions?

File: lib/testing/src/lib/content-services/pages/siteDropdown.page.ts [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    sitesDropdown;
+    dropDownPanel = element(by.css(`div[class*='mat-select-panel']`));
+
+    constructor(rootElement: ElementFinder = element(by.css(`adf-sites-dropdown`))) {
+        this.rootElement = rootElement;
+        this.sitesDropdown = this.rootElement.element(by.css(`mat-select[data-automation-id='site-my-files-option']`));
+    }
+
+    selectLocation(option) {
+        BrowserVisibility.waitUntilElementIsVisible(this.sitesDropdown);
```
Comment (eromano) [inline lib/testing/src/lib/content-services/pages/siteDropdown.page.ts]: ```BrowserActions.click``` instead of  
```
        BrowserVisibility.waitUntilElementIsVisible(this.sitesDropdown);
        this.sitesDropdown.click();
```

File: lib/testing/src/lib/core/pages/data-table-component.page.ts [hunk 9]
```diff
@@ -300,6 +300,16 @@ export class DataTableComponentPage {
         return row.click();
     }
 
+    doubleClickRowByContent(content) {
+        const row = this.getCellByContent(content);
+        BrowserVisibility.waitUntilElementIsVisible(row);
+        BrowserVisibility.waitUntilElementIsClickable(row);
+        row.click();
```
Comment (eromano) [inline lib/testing/src/lib/core/pages/data-table-component.page.ts]: can we just use BrowserActions.click(row) instead of 

```
BrowserVisibility.waitUntilElementIsVisible(row);
        BrowserVisibility.waitUntilElementIsClickable(row);
        row.click();
```

File: e2e/content-services/custom-site-list.e2e.ts (hunk 7 above)
Comment (cristinaj) [reply e2e/content-services/custom-site-list.e2e.ts]: My opinion is that moving the tests in a separate PR we are increasing the number of blocked PRs, plus it might be complicated at rebase.

Comment (gmandakini) [reply e2e/content-services/custom-site-list.e2e.ts]: That was my concern as well. If we keep it in separate PR, we might as well forget about it. It will be better done fresh, when the bug is fixed. Rebasing is a pain. but I had a conversation with Eugenio, and this is what we have agreed. maybe worth having a discussion in retro.

File: e2e/content-services/custom-site-list.e2e.ts (hunk 1 above)
Comment (adinapitul) [reply e2e/content-services/custom-site-list.e2e.ts]: this shouldn't be needed, all involved pages here are angular.
try using `await loginPage.loginToContentServicesUsingUserModel(acsUser);`

**Comment by mauriziovitale** [comment]
Very old PR in the case is needed reopen it
